go/private/repository_tools.bzl

```diff
@@ -76,23 +30,34 @@ filegroup(
 """
 
 def _go_repository_tools_impl(ctx):
-  go_tool = ctx.path(ctx.attr._go_tool)
-  goroot = go_tool.dirname.dirname
-  gopath = ctx.path('')
-  prefix = "github.com/bazelbuild/rules_go/" + ctx.attr._tools.package
-  src_path = ctx.path(ctx.attr._tools).dirname
-
-  _fetch_repository_tools_deps(ctx, goroot, gopath)
-
-  for t, pkg in [("gazelle", 'gazelle/gazelle'), ("fetch_repo", "fetch_repo")]:
-    ctx.symlink("%s/%s" % (src_path, t), "src/%s/%s" % (prefix, t))
+  # We have to download this directly because the normal version is based on go_repository
+  # and thus requires the gazelle we build in here to generate it's BUILD files
+  # The commit used here should match the one in repositories.bzl
```

Comment:
I have deduplicated it within this file, but I don't want to share it will the main repository rules, because it does not actually matter if they are different, and I don't want a dependency between the two. The ones in repository.bzl are designed to move into the workspace as soon as nested workspaces arrive.